### PR TITLE
Android filesystem fix

### PR DESCRIPTION
**Pull request by FalseIlyu**
Solves #687

This solves three problem i had when launching the game which are, crash with SDL (corrected by updating SDL java interface), changing the JavaFilesystem recursive directory search as it doesn't give the file path relative to the first directory meaning the path are wrong when trying to open them, finally i openned the soundfile using `soundPack.Open(fileSystem.ReadAll(f));` as the normal open use std::filesystem to open files.

There also seems to be a problem with the game path but my installation is non standard and is inside the user document folder instead of /data/local/tmp/bw.

Due to this fact and until i have test feedback on other android systems this PR shall be kept a draft.

An other prerequisite is investigating this \:
image

### CODE REVIEW

File: src/Game.cpp [hunk 1]
```diff
@@ -667,15 +669,15 @@
...
 			return;
 		}
 
 		pack::PackFile soundPack;
 		SPDLOG_LOGGER_DEBUG(spdlog::get("audio"), "Opening sound pack {}", f.filename().string());
-		soundPack.Open(f);
+		soundPack.Open(fileSystem.ReadAll(f));
 		const auto& audioHeaders = soundPack.GetAudioSampleHeaders();
 		const auto& audioData = soundPack.GetAudioSamplesData();
 		auto soundName = std::filesystem::path(audioHeaders[0].name.data());
```
[the remark this answers is not in the document]
Comment (bwrsandman) [reply src/Game.cpp:683]: Not sure what this correction is about

File: android/app/src/main/java/org/libsdl/app/HIDDeviceBLESteamController.java [hunk 2]
```diff
@@ -76,11 +76,10 @@ private GattOperation(BluetoothGatt gatt, GattOperation.Operation operation, UUI
...
-                // This is executed in main thread
-                BluetoothGattCharacteristic chr;
+            // This is executed in main thread
+            BluetoothGattCharacteristic chr;
 
-                switch (mOp) {
+            switch (mOp) {
                 case CHR_READ:
                     chr = getCharacteristic(mUuid);
                     //Log.v(TAG, "Reading characteristic " + chr.getUuid());
```
Comment (bwrsandman) [inline android/app/src/main/java/org/libsdl/app/HIDDeviceBLESteamController.java:85]:
```
 /home/runner/work/openblack/openblack/android/app/src/main/java/org/libsdl/app/HIDDeviceBLESteamController.java:86: Error: Call requires permission which may be rejected by user: code should explicitly check to see if permission is available (with checkPermission) or explicitly handle a potential SecurityException [MissingPermission]
                    if (!mGatt.readCharacteristic(chr)) {
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Running `gradle lint` will outline the errors

**Comment by FalseIlyu** [comment]
I manage to run the game on emulator with this changes, run test on other android emulator and real android platform are i think necessary before marking it as ready.
The graphical problem still needs investigation but unless found and solved in the following days i think it should be the purview of another PR.

**Comment by bwrsandman** [comment]
I tried out the emulator for a pixel 
image

**Comment by bwrsandman** [comment]
Tried on 3 physical devices all arm:
* Pixel 7 Pro with Android 14 arm64
* Galaxy Tablet SM-X700 with Android 14 arm64
* Quest 3 with Android 12L on arm64

Unfortunately all three of them had a black screen
image



I am trying without success to emulate an arm64 device in android studio.

**Comment by bwrsandman** [comment]
Running in renderdoc shows that it never presents

**Review by bwrsandman — APPROVED** [review]

**Pull request merged**